### PR TITLE
add vicreg, add cifar-10, refactor, small fixes

### PR DESCRIPTION
**Pull request by ekitanidis**
- Add VICReg and refactor methods and model params
- Add CIFAR-10 and refactor datasets and data transforms
- Make LARS eta into hyperparam
- Add more logging
- Fix bug from pytorch-lightning versions >= 1.3.0
- Update README

### CODE REVIEW

File: moco.py [hunk 1]
```diff
@@ -112,7 +56,10 @@ def __init__(
         elif isinstance(hparams, dict):
             hparams = self.params(**hparams, **kwargs)
 
-        self.hparams = AttributeDict(attr.asdict(hparams))
+        if pl.__version__ >= '1.3.0':
```
Comment (abefetterman) [inline moco.py:59]: I'm not sure this comparison is the right way to check the version, perhaps use 

`if isinstance(self.hparams, AttributeDict):`

File: README.md [hunk 2]
```diff
@@ -16,114 +23,119 @@ cd self_supervised
...
+os.environ["DATA_PATH"] = "~/data"
+
+params = SimCLRParams()
+model = SelfSupervisedMethod(params)
+trainer = pl.Trainer(gpus=1, max_epochs=320)
+trainer.fit(model)
+trainer.save_checkpoint("example.ckpt")
 ```
 
 This is not super efficient as it still has the separate offline network, and so it does each embedding calculation twice. However, it is sufficient for comparing the methods with each other.
```
Comment (abefetterman) [inline README.md:112]: Can delete this line, we have no separate offline network for SimCLR now

**Review by abefetterman — COMMENTED** [review]
It's looking great! I had only a few things to check, and we should confirm running SimCLR works properly after the update.

File: moco.py (hunk 1 above)
Comment (ekitanidis) [reply moco.py]: Good point, string comparison is too brittle. I added your fix, which works for all versions of pytorch-lightning I tested.

File: README.md (hunk 2 above)
Comment (ekitanidis) [reply README.md]: Done.

File: model_params.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    loss_constant_factor=2,
+)
+
+SimCLRParams = partial(
+    ModelParams,
+    use_negative_examples_from_batch=True,
+    use_negative_examples_from_queue=False,
+    K=0,
+    m=0.0,
+    use_both_augmentations_as_queries=True,
```
[the remark this answers is not in the document]
Comment (ekitanidis) [reply model_params.py:119]: Done, and tested that SimCLR seems to work.

**Pull request merged**